Index: debian/scripts/gen_build_files.sh
===================================================================
--- debian.orig/scripts/gen_build_files.sh	2011-02-14 21:55:21.732103919 +0300
+++ debian/scripts/gen_build_files.sh	2011-02-14 21:55:26.878008920 +0300
@@ -59,7 +59,7 @@
 	"${s}"
 
 # (Re)generate */Kbuild and */Config.in
-{ cd -- "$srctree" && find . -type d; } | while read -r d; do
+{ cd -- "$srctree" && find . -name .\?\* -prune -or -type d -print; } | while read -r d; do
 	d="${d#./}"
 
 	src="$srctree/$d/Kbuild.src"
